apps/ta_dump.py

```diff
@@ -87,67 +101,94 @@ def algorithm_hist(algorithms):
     plt.legend()
 
     plt.tight_layout()
-    plt.savefig("algorithm_histogram.svg")
+    plt.savefig("ta_algorithm_histogram.svg")
     plt.close()
 
 def det_type_hist(det_types):
     """
     Plot a histogram of the detector type for the TAs.
     """
     plt.figure(figsize=(12,8))
-    plt.hist(np.array(det_types).flatten(), bins=np.arange(-0.5, 3, 1), range=(0,2), align='mid', color='k')
+    plt.hist(det_types, bins=np.arange(-0.5, 3, 1), range=(0,2), align='mid', color='k')
 
     plt.title("TA Detector Type Histogram")
     plt.xticks(ticks=range(0,3), labels=("Unknown",
                                          "TPC",
                                          "PDS"), rotation=60)
 
     plt.tight_layout()
-    plt.savefig("det_types_histogram.svg")
+    plt.savefig("ta_det_types_histogram.svg")
     plt.close()
 
 def adc_integral_hist(adc_integrals):
     """
     Plot a histogram of the ADC integrals for the TAs.
     """
     plt.figure(figsize=(6,4))
-    plt.hist(np.array(adc_integrals).flatten(), color='k')
+    plt.hist(adc_integrals, color='k')
 
     plt.title("TA ADC Integral Histogram")
     plt.xlabel("ADC Integral")
 
     plt.tight_layout()
-    plt.savefig("adc_integral_histogram.svg")
+    plt.savefig("ta_adc_integral_histogram.svg")
     plt.close()
 
-def plot_time_window_summary(ta_data, tp_data, quiet=False):
+def plot_time_window_summary(ta_data, tp_data, quiet=False, seconds=False):
     """
     Plot summary statistics on time windows.
 
     This uses 2 definitions for time windows:
         Direct Diff: max(tp.time_start) - ta.time_start,
         Max Diff: max(tp.time_start + tp.time_over_threshold) - ta.time_start
     """
-    direct_diff = []
-    max_diff = []
+    time_unit = 's' if seconds else 'Ticks'
 
-    for ta_frag, tp_frag in zip(ta_data, tp_data):
-        for ta, tps in zip(ta_frag, tp_frag):
-            direct_diff += list(np.max(tps['time_start']) - ta['time_start'])
-            max_diff += list(np.max(tps['time_start'] + tps['time_over_threshold']) - ta['time_start'])
-    direct_diff = np.array(direct_diff)
-    max_diff = np.array(max_diff)
+    direct_diff = np.zeros(ta_data.shape)
+    max_diff = np.zeros(ta_data.shape)
+
+    # Find the differences for each TA
+    for idx in np.arange(ta_data.shape[0]):
+        direct_diff[idx] = np.max(tp_data[idx]['time_start']) - ta_data[idx]['time_start']
+        max_diff[idx] = np.max(tp_data[idx]['time_start'] + tp_data[idx]['time_over_threshold']) - ta_data[idx]['time_start']
+
+    if seconds:
+        direct_diff = direct_diff * TICK_TO_SEC_SCALE
+        max_diff = max_diff * TICK_TO_SEC_SCALE
```

Comment:
NICE!
I imagine it could be even faster if the [idx]['time_start'] were transposed, there would be no need for the for loop and numpy would do that as an optimized matrix operation -- but that's maybe for the further future.

Reply:
The issue occurs with `tp_data`. Since the amount of TPs per TA is irregular, this one has to be a list of NumPy arrays. I don't immediately know how this could be further optimized.

Reply:
Ah yes, this is ta dump not tp dump... nvm